package-installer/src/main/java/com/vmware/pscoe/iac/installer/Installer.java

```diff
@@ -1380,68 +1385,124 @@ private static List<Package> getFilesystemPackages(final PackageType type) {
+            if (input.allTrue(Option.VRO_EMBEDDED)) {
+                prefixes.add(ConfigurationPrefix.VRO.getValue());
+                prefixes.add(ConfigurationPrefix.VRANG.getValue());
+            } else {
+                prefixes.add(ConfigurationPrefix.VRO.getValue());
+            }
+            RestClientVro client = RestClientFactory.getClientVro(ConfigurationVro.fromProperties(input.getMappings(prefixes.toArray(new String[0]))));
+            WorkflowExecution workflowExecutionResult = new VroWorkflowExecutor(client).executeWorkflow(wfID, wfInput, Integer.parseInt(wfTimeout));
+
+            JsonObject wfOutputJson = new JsonObject();
+            for (String key : workflowExecutionResult.getOutput().stringPropertyNames()) {
+                String value = StringEscapeUtils.unescapeJson(workflowExecutionResult.getOutput().getProperty(key));
+                wfOutputJson.add(key, gson.fromJson(value, JsonObject.class));
+            }
+            // write the workflow output
+            String outputString = gson.toJson(wfOutputJson);
+            if (Installer.endsWithIgnoreCase(wfOutputFilePath, "yaml")) {
+                outputString = Installer.convertJsonObjectToYaml(wfOutputFilePath, wfOutputJson);
+            }
+
+            Files.write(Paths.get(wfOutputFilePath), outputString.getBytes(), StandardOpenOption.CREATE);
+            // write the workflow error in a workflow error file (if any)
+            if (!StringUtils.isEmpty(workflowExecutionResult.getError())) {
+                Files.write(Paths.get(wfErrFilePath), workflowExecutionResult.getError().getBytes(), StandardOpenOption.CREATE);
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Unable to perform file operation: " + e.getClass().getName() + " : " + e.getLocalizedMessage(), e);
+        } catch (JsonSyntaxException e) {
+            throw new RuntimeException("Unable to parse input / output file data: " + e.getClass().getName() + " : " + e.getLocalizedMessage(), e);
+        } catch (WorkflowExecutionException e) {
+            try {
+                Files.write(Paths.get(wfErrFilePath), e.getMessage().getBytes(), StandardOpenOption.CREATE);
+            } catch (IOException e1) {
+                throw new RuntimeException("Unable to perform file operation: " + e.getClass().getName() + " : " + e.getLocalizedMessage(), e);
+            }
+            throw new RuntimeException("Workflow execution failed: " + e.getClass().getName() + " : " + e.getLocalizedMessage(), e);
+        } catch (Exception e) {
+            throw new RuntimeException("General error during workflow execution: " + e.getClass().getName() + " : " + e.getLocalizedMessage(), e);
+        }
+    }
+	
+    private static boolean isValidYaml(String yamlString) {
+        Yaml yaml = new Yaml();
+        try {
+            yaml.load(yamlString);
+            return true;
+        } catch (ParserException e) {
```

Comment:
I agree here, I think it would make sense to log this error, I did a quick check and it seems our convention for similar error handlings is to use `error` severity of logger to log the error message or a custom message. Maybe it would be beneficial to combine both, something like:
`Encountered the following error while parsing Installer workflow input as yaml: + the error message itself`

Reply:
Discussed offline that it makes more sense to tackle this as a debug message otherwise on every valid JSON configuration we will have error messages in the logs.

Reply:
Code updated to have a debug message.